<launch>
    <rosparam param="/total_agents"> 5 </rosparam>

    <group ns="agent0">
        <node name="RoleManager0" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="0 5 5">
        </node>
        <node name="DataCenter0" pkg="Voronoi_Based_CBSA" type="data_center.py" output="screen" args="0">
        </node>
    </group>

    <group ns="agent1">
        <node name="RoleManager1" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="1 15 5">
        </node>
        <node name="DataCenter1" pkg="Voronoi_Based_CBSA" type="data_center.py" output="screen" args="1">
        </node>
    </group>

    <group ns="agent2">
        <node name="RoleManager2" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="2 5 15">
        </node>
        <node name="DataCenter2" pkg="Voronoi_Based_CBSA" type="data_center.py" output="screen" args="2">
        </node>
    </group>

    <group ns="agent3">
        <node name="RoleManager3" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="3 10 10">
        </node>
        <node name="DataCenter3" pkg="Voronoi_Based_CBSA" type="data_center.py" output="screen" args="3">
        </node>
    </group>

    <group ns="agent4">
        <node name="RoleManager4" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="4 15 15">
        </node>
        <node name="DataCenter4" pkg="Voronoi_Based_CBSA" type="data_center.py" output="screen" args="4">
        </node>
    </group>
<!--
    <group ns="agent5">
        <node name="RoleManager5" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="5 3 8">
        </node>
        <node name="DataCenter5" pkg="Voronoi_Based_CBSA" type="data_center.py" output="screen" args="5">
        </node>
    </group>

    <group ns="agent6">
        <node name="RoleManager6" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="6 17 9">
        </node>
        <node name="DataCenter6" pkg="Voronoi_Based_CBSA" type="data_center.py" output="screen" args="6">
        </node>
    </group>

    <group ns="agent7">
        <node name="RoleManager7" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="7 11 14.7">
        </node>
        <node name="DataCenter7" pkg="Voronoi_Based_CBSA" type="data_center.py" output="screen" args="7">
        </node>
    </group>

    <group ns="agent8">
        <node name="RoleManager8" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="8 6 17">
        </node>
        <node name="DataCenter8" pkg="Voronoi_Based_CBSA" type="data_center.py" output="screen" args="8">
        </node>
    </group>

    <group ns="agent9">
        <node name="RoleManager9" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="9 17 13">
        </node>
        <node name="DataCenter9" pkg="Voronoi_Based_CBSA" type="data_center.py" output="screen" args="9">
        </node>
    </group>
   -->
    <!--
    <node name="agent9" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="9 17 13">
    </node>

    <node name="agent10" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="10 13 18">
    </node>

    <node name="agent11" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="11 6 19">
    </node>

    <node name="agent12" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="12 17 11">
    </node>

    <node name="agent13" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="13 16 4">
    </node>

    <node name="agent14" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="14 9 15">
    </node>

    <node name="agent15" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="15 19.66 19.63">
    </node>

    <node name="agent16" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="16 11.89 14.96">
    </node>

    <node name="agent17" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="17 0.98 19.48">
    </node>

    <node name="agent18" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="18 14.43 5.57">
    </node>

    <node name="agent19" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="19 8.3 8.14">
    </node> -->
<!-- 
    <node name="agent20" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="20">
    </node>

    <node name="agent21" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="21">
    </node>

    <node name="agent22" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="22">
    </node>

    <node name="agent23" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="23">
    </node>

    <node name="agent24" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="24">
    </node>

    <node name="agent25" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="25">
    </node>

    <node name="agent26" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="26">
    </node>

    <node name="agent27" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="27">
    </node>

    <node name="agent28" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="28">
    </node>

    <node name="agent29" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="29">
    </node>

    <node name="agent30" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="30">
    </node>

    <node name="agent31" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="31">
    </node>

    <node name="agent32" pkg="Voronoi_Based_CBSA" type="agent.py" output="screen" args="32">
    </node> -->
</launch>
